server/chat/background/task.py

```
@@ -455,6 +483,28 @@ def run_background_chat(
     logger.info(f"[BackgroundChat] Starting for user {user_id}, session {session_id}")
     logger.info(f"[BackgroundChat] Trigger: {trigger_metadata}")
 
+    # Deduplication guard: acks_late=True can cause Redis to redeliver the same task
+    # to the worker while it's still running. Use a Redis SETNX lock on the task ID.
+    _dedup_key = f"celery:dedup:{self.request.id}"
+    try:
+        _redis = celery_app.backend.client
+        _acquired = _redis.set(_dedup_key, "1", nx=True, ex=1800)
+        if not _acquired:
+            logger.warning(f"[BackgroundChat] DEDUP: Task {self.request.id} already running, skipping duplicate execution")
```

Ruff (0.15.17)

[warning] Logging statement uses f-string

(G004)